server/game/cards/characters/05/tommenbaratheon.js

```diff
@@ -0,0 +1,32 @@
+const DrawCard = require('../../../drawcard.js');
+
+class TommenBaratheon extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => this.controller.hand.size() === 0,
+            targetType: 'player',
+            effect: ability.effects.cannotGainChallengeBonus()
+        });
+
+        this.persistentEffect({
+            condition: () => this.opponentHasNoCardsInHand(),
```

Comment:
The following should be equivalent to your two separate effects and will get rid of the extra function:
```javascript
this.persistentEffect({
  targetType: 'player',
  targetController: 'any',
  match: player => player.hand.size() === 0,
  effect: ability.effects.cannotGainChallengeBonus()
});
```